### PR TITLE
py/asmthumb: Provide implementations of clz/ctz for MSC.

### PR DESCRIPTION
**Pull request by dpgeorge**
This is needed since a5324a10747dfba921bb363ed904f05ebb6de010

@stinos is this the correct way to do it?

### CODE REVIEW

File: py/asmthumb.c [hunk 1]
```diff
@@ -36,6 +36,23 @@
 #include "py/mpstate.h"
 #include "py/asmthumb.h"
 
+#ifdef _MSC_VER
+#include <intrin.h>
+
+static uint32_t mp_clz(uint32_t x) {
+    unsigned long lz = 0;
+    return _BitScanReverse(&lz, x) ? 31 - lz : 0;
```
Comment (stinos) [inline py/asmthumb.c:44]: `(sizeof(x) * 8 - 1) instead of `31` , or is that overkill ?

**Comment by stinos** [comment]
Looks correct yes. Better to use 'msvc' instead of 'MSC' in commit message because former is used more often generally, and all other commits use it as well.

But my first idea was: does it even make sense to compile this whole file with MSVC?

**Comment by dpgeorge** [comment]
> But my first idea was: does it even make sense to compile this whole file with MSVC?

Yes I think so, it's building mpy-cross and allows the user to cross-compile ARMv6-M native code on Windows.

**Comment by stinos** [comment]
Ok, right. Is this in the test suite somewhere?

**Comment by dpgeorge** [comment]
> Is this in the test suite somewhere?

Not exactly... mpy-cross is used this way in the qemu-arm CI to compile native code to ARMv7-M, and in the unix tests to compile to x86/x64.  But there's no specific test to run for mpy-cross to build all the architectures it knows about.

**Comment by stinos** [comment]
Not a big deal probably, so this is good to go then.

File: py/asmthumb.c (hunk 1 above)
Comment (dpgeorge) [reply py/asmthumb.c]: done

**Comment by dpgeorge** [comment]
> Better to use 'msvc' instead of 'MSC' in commit message because former is used more often generally, and all other commits use it as well.

Done.

**Pull request merged**